pkg/util/arm/arm.go

```diff
@@ -1,128 +1,223 @@
 package arm
 
 import (
-	"fmt"
-	"sort"
-	"strings"
-
-	armconst "github.com/openshift/openshift-azure/pkg/arm/constants"
-	"github.com/openshift/openshift-azure/pkg/util/jsonpath"
-	"github.com/openshift/openshift-azure/pkg/util/resourceid"
+	"encoding/json"
+	"math"
+	"reflect"
 )
 
 type Template struct {
-	Schema         string        `json:"$schema,omitempty"`
-	ContentVersion string        `json:"contentVersion,omitempty"`
-	Parameters     struct{}      `json:"parameters,omitempty"`
-	Variables      struct{}      `json:"variables,omitempty"`
-	Resources      []interface{} `json:"resources,omitempty"`
-	Outputs        struct{}      `json:"outputs,omitempty"`
+	Schema         string                 `json:"$schema,omitempty"`
+	APIProfile     string                 `json:"apiProfile,omitempty"`
+	ContentVersion string                 `json:"contentVersion,omitempty"`
+	Variables      map[string]interface{} `json:"variables,omitempty"`
+	Parameters     map[string]Parameter   `json:"parameters,omitempty"`
+	Functions      []interface{}          `json:"functions,omitempty"`
+	Resources      []Resource             `json:"resources,omitempty"`
+	Outputs        map[string]interface{} `json:"outputs,omitempty"`
 }
 
-// FixupAPIVersions inserts an apiVersion field into the ARM template for each
-// resource (the field is missing from the internal Azure type).
-func FixupAPIVersions(template map[string]interface{}, versions map[string]string) error {
```

Comment:
I am unconvinced that we should remove FixupAPIVersions or FixupDepends, at the very least in this PR - let's discuss.